solidity/app/contracts/Router.sol

```diff
@@ -130,22 +132,50 @@ abstract contract Router is AbacusConnectionClient, IMessageRecipient {
     }
 
     /**
-     * @notice Dispatches a message to an enrolled router via the local router's
-     * Outbox and pays for message processing on the destination chain.
-     * @dev Reverts if there is no enrolled router for _destination.
+     * @notice Pay for message processing on the destination
+     * @param _leafIndex The leaf index of the message to pay processing for
+     * @param _gasPayment The amount of native tokens to pay the Interchain Gas
+     * Paymaster to process the dispatched message.
+     */
+    function _payForGas(uint256 _leafIndex, uint256 _gasPayment) internal {
+        _interchainGasPaymaster().payGasFor{value: _gasPayment}(_leafIndex);
+    }
+
+    /**
+     * @notice Calls #checkpoint on the Outbox
+     */
+    function _checkpointOnOutbox() internal {
+        _outbox().checkpoint();
+    }
+
+    /**
+     * @notice A convenience function which allows the caller to 1) send a message
+     * to the enrolled router on _destination as specified on the ACM, 2) pay
+     * for message processing on the destination chain and 3) checkpoint the
+     * message on the Outbox
      * @param _destination The domain of the chain to which to send the message.
      * @param _msg The message to dispatch.
      * @param _gasPayment The amount of native tokens to pay the Interchain Gas
      * Paymaster to process the dispatched message.
+     * @param _shouldCheckpoint Whether checkpoint should be called on the Outbox
      */
-    function _dispatchToRemoteRouterWithGas(
+    function _comboDispatch(
```

Comment:
`comboDispatch` needs a better name.

Consider `dispatchMessage` and `dispatchPaidMessage` in place of `dispatchToRemoteRouter` and `comboDispatch`, respectively...

Alternatively consider `dispatch` and `dispatchPaid` or `dispatch` and `paidDispatch`

Reply:
Indeed, `comboDispatch`. I wasn't a fan of including either the notion of gas payment or checkpointing in the name since that is optional and depends on the passed in parameters. I'm inclined to just have `comboDispatch` to be just `dispatch`, even though I'm not generally a fan of a wrapper function to be identically named after a strict subset of functionality